std/machines/large_field/arith256_memory.asm

```diff
@@ -218,61 +212,85 @@ machine Arith256Memory(mem: Memory) with
+        |values| match limbs_to_ints(values) {
+            [y2, y3, x1] => {
+                let dividend = (y2 << 256) + y3;
+                int_to_limbs(dividend / x1) + int_to_limbs(dividend % x1)
+            },
+            _ => panic("Unexpected number of values")
+        }
+    );
 
-    let provide_values = query |column_arr, row, value| {
-        let _ = array::map_enumerated(column_arr, |j, column| std::prover::provide_value(column, row, fe(select_limb(value, j))));
-    };
-    query |i| {
-        let op = get_operation();
-        if op == "affine_256" || op == "mod_256" {
-            match std::prover::try_eval(y1[0]) {
-                Option::Some(_) => {
-                    // y1 is an input, in this case we do not need a hint.
-                },
-                Option::None => {
-                    // y1 is not an input, which means we are probably computing
-                    // division or modulo.
-                    let y2 = y2_int();
-                    let y3 = y3_int();
-                    let x1 = x1_int();
-                    let dividend = (y2 << 256) + y3;
-                    provide_values(y1, i, dividend / x1);
-                    provide_values(x2, i, dividend % x1);
-                }
-            }
-        } else {
-            let y1 = y1_int();
-            // y2 is unused for ec_double
-            let y2 = if op == "ec_add" { y2_int() } else { 0 };
-            let x1 = x1_int();
-            let x2 = x2_int();
-            let s_val = if op == "ec_add" {
-                div(sub(y2, y1), sub(x2, x1))
-            } else {
-                div(mul(3, mul(x1, x1)), mul(2, y1))
-            };
-            provide_values(s, i, s_val);
-            // Compute quotients.
-            // Note that we add 2**258 to it, to move it from the (-2**258, 2**258) to the (0, 2**259) range, so it can
-            // be represented as an unsigned 272-bit integer.
-            // See the comment for `product_with_p` below.
-            let q0_val = if op == "ec_add" {
-                -(s_val * x2 - s_val * x1 - y2 + y1) / secp_modulus + (1 << 258)
-            } else {
-                -(2 * s_val * y1 - 3 * x1 * x1) / secp_modulus + (1 << 258)
-            };
-            provide_values(q0, i, q0_val);
-
-            // Adding secp_modulus to make sure that all numbers are positive when % is applied to it.
-            let x3_val = (s_val * s_val - x1 - x2 + 2 * secp_modulus) % secp_modulus;
-            provide_values(x3, i, x3_val);
-            let y3_val = (s_val * ((x1 - x3_val) + secp_modulus) - y1 + secp_modulus) % secp_modulus;
-            provide_values(y3, i, y3_val);
-
-            provide_values(q1, i, -(s_val * s_val - x1 - x2 - x3_val) / secp_modulus + (1 << 258));
-            provide_values(q2, i, -(s_val * x1 - s_val * x3_val - y1 - y3_val) / secp_modulus + (1 << 258));
+    // Prover function for mod_256
+    query |i| std::prover::compute_from_multi_if(
+        operation_id = 2,
```

Comment:
We don't have that in the other file - did I miss it?

Reply:
In arith, affine_256 and mod_256 share the same operation id, so maybe you shared the prover function there as well? In this machine, the operation ids are different.